src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProcessor.cs

```diff
@@ -25,7 +25,9 @@ public ConsoleLoggerProcessor()
                 IsBackground = true,
                 Name = "Console logger queue processing thread"
             };
+#pragma warning disable CA1416 // Validate platform compatibility, not sure if this assembly is used for browser, suppressing for now
             _outputThread.Start();
```

Comment:
Maybe we should just mark Console Logging as unsupported on Browser.

Cc @maryamariyan @tarekgh

Reply:
OK, i guess that is the real fix, i will update

Reply:
We could someday make it work on Browser, but that would take more effort than I think we want to do in this PR. And someone would need to have a scenario/use case for it. AFAIK, Blazor WASM doesn't use the Console Logger and I haven't seen anyone asking for it.

Reply:
Blazor WebAssembly uses this Logger by default:  https://github.com/dotnet/aspnetcore/blob/63199b40afd17586ee25d4ab29a42af5058edac9/src/Components/WebAssembly/WebAssembly/src/Services/WebAssemblyConsoleLogger.cs

https://github.com/dotnet/aspnetcore/blob/63199b40afd17586ee25d4ab29a42af5058edac9/src/Components/WebAssembly/WebAssembly/src/Services/WebAssemblyConsoleLoggerProvider.cs

https://github.com/dotnet/aspnetcore/blob/63199b40afd17586ee25d4ab29a42af5058edac9/src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostBuilder.cs#L230

Reply:
Thanks, @campersau, form the links I don't see they are using Microsoft.Extensions.Logging.Console, so i assume it is confirming that marking the entire assembly Unsupported on browser is safe

> We could someday make it work on Browser, but that would take more effort than I think we want to do in this PR. And someone would need to have a scenario/use case for it. AFAIK, Blazor WASM doesn't use the Console Logger and I haven't seen anyone asking for it.

From the link above i see it uses `WebAssemblyJSRuntime` or Javascript console which makes sense for browser app, i don't imagine browser would ever need/could access command-line console for logging. But just in case i can try to annotate only the types involved in calling unsupported APIs first instead of adding assembly-level unsupported attribute

Reply:
I think the only thing that doesn't work here is ConsoleLoggerProcessor?

Reply:
> I think the only thing that doesn't work here is ConsoleLoggerProcessor?

ConsoleLoggerProcessor is at the very core of the library. It is the thing that actually writes the log message to the `System.Console` (batched on a dedicated thread - which is the unsupported API). Without it, the rest of the library is useless.

Reply:
Sorry I was trying to suggest that marking options classes as unsupported can cause problems where it doesn't need to.

Reply:
Attributed only affected types for now

Reply:
After looking at your latest changes, I'm wondering "Is there a bug in the analyzer here?".  You marked `ConsoleLoggerProvider` as `[UnsupportedOSPlatform("browser")]`, but there are still usages of the type that aren't raising analysis warnings:

https://github.com/dotnet/runtime/blob/e82f4af06623fb489e2a55b10e16c5fb20d373cb/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerExtensions.cs#L29-L30

The above is how the `ConsoleLoggerProvider` class gets instantiated - through DI.

The type is also used here:

https://github.com/dotnet/runtime/blob/e82f4af06623fb489e2a55b10e16c5fb20d373cb/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerExtensions.cs#L157-L175

Reply:
> The above is how the ConsoleLoggerProvider class gets instantiated - through DI.

> The type is also used here:

Good catch, those are not a covered and not considered scenario. To cover both could we warn for any type of parameter usages? If that could cause any false positives?

Reply:
> To cover both could we warn for any type of parameter usages? If that could cause any false positives?

My understanding of the intention here:

1. The whole class/type is marked as `[UnsupportedOSPlatform("browser")]`, meaning that "this class is not going to work on that platform".
2. The method call `SomeGenericMethod<AboveType>()` should warn about using this type. It is not supported on this platform.

I guess I don't see how a false positive could occur here. You are using a type that is not supported on the platform. It seems like it should warn.

Reply:
Created an [issue](https://github.com/dotnet/roslyn-analyzers/issues/4362) in roslyn-analzyers repo

Reply:
We decided to warn when generic platform-specific types used in operation blocks/executable code, but will not flag/warn the uses of platform-specific types in declarations. That means https://github.com/dotnet/runtime/blob/e82f4af06623fb489e2a55b10e16c5fb20d373cb/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerExtensions.cs#L29-L30

Will be warned, but the declarations below will not be warned
https://github.com/dotnet/runtime/blob/e82f4af06623fb489e2a55b10e16c5fb20d373cb/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerExtensions.cs#L157-L175

The bug fix didn't found any other browser unsupported warning except the one above, i have fixed that warning, please review